drivers/pwm/pwm_nrfx.c

```diff
@@ -315,20 +381,24 @@ static void pwm_suspend(const struct device *dev)
 
 	memset(dev->data, 0, sizeof(struct pwm_nrfx_data));
 	(void)pinctrl_apply_state(config->pcfg, PINCTRL_STATE_SLEEP);
+
+	return 0;
 }
 
 static int pwm_nrfx_pm_action(const struct device *dev,
 			      enum pm_device_action action)
 {
+	int ret = 0;
+
 	if (action == PM_DEVICE_ACTION_RESUME) {
-		pwm_resume(dev);
+		ret = pwm_resume(dev);
 	} else if (IS_ENABLED(CONFIG_PM_DEVICE) && (action == PM_DEVICE_ACTION_SUSPEND)) {
-		pwm_suspend(dev);
+		ret = pwm_suspend(dev);
```

Comment:
Why not just add `return` in these two lines (and remove that unneeded then `return 0` at the of the function) instead of introducing the `ret` variable?